gremlin-dotnet/src/Gremlin.Net/Process/Traversal/GraphTraversal.cs

```diff
@@ -772,8 +772,16 @@ private GraphTraversal(ICollection<ITraversalStrategy> traversalStrategies, Byte
         /// </summary>
         public GraphTraversal<S, E> HasId (object id, params object[] otherIds)
         {
-            var args = new List<object>(1 + otherIds.Length) {id};
```

Comment:
@mikepersonick This change is basically the same you did in #1557 for the `P` predicates. We have other steps here that also take a `params` array argument and then access its `.Length` without checking whether it's `null` first (e.g., the `HasKey` step below).
As I mentioned in the PR description, I'm not sure whether we want to allow `null` everywhere or maybe throw an `ArgumentNullException` in some places where we're sure that `null` isn't a valid argument.
Any thoughts on that?

Otherwise, I'd suggest that we create a ticket for this issue and then address it in a follow-up PR where we can concentrate on just that topic.

Reply:
Florian, thanks the review and merge on the other PR. I’m out today so I’ll take a closer look on Monday, but I guess my general feeling is that whatever the semantics are on `null` (or anything else), it’s probably best to handle them in the query execution layer rather than the language bindings.

Reply:
Thanks for your input. I've created an issue for that: https://issues.apache.org/jira/browse/TINKERPOP-2705
because it warrants its own issue & PR in my opinion so we can concentrate only on that topic.